Sorting based on flagging status

helihard opened the pull request:
## Description
This PR enables sorting based on flagging status (action needed, action taken, empty) in the 'Organizer action needed' table.


## Screenshots
![asc](https://github.com/user-attachments/assets/654f73e7-761b-467a-8db4-88de313ff2b7)
![desc](https://github.com/user-attachments/assets/54d0a1f4-9109-479e-87b2-066135300dd8)
![none](https://github.com/user-attachments/assets/1543aa64-8ee5-4fda-8e16-2da63d8af3f5)

## Changes
* Adds a sortComparator to the 'organizer action' column type. Rows are sorted based on flagging status:
Highest priority: there are flagged calls (organizer_action_needed), and at least one call has an unresolved issue (organizer action has not been taken)
Medium: there are flagged calls, and none of them has an unresolved issue
Lowest: empty/white


## Notes to reviewer
I am a bit confused regarding what would be considered 'asc' or 'desc' sorting in this case, but that could easily be altered in the code.


## Related issues
Resolves #2474

ziggabyte reviewed (dismissed):
I am approving this, but also giving a suggestion for an improvement for making the code a bit more consise. You decide if you wanna make a new commit with this, or leave it :)

richardolsson commented on src/features/views/components/ViewDataTable/columnTypes/OrganizerActionColumnType.tsx:
```diff
@@ -27,6 +27,27 @@ import useToggleDebounce from 'utils/hooks/useToggleDebounce';
+
+  const getSortValue = (v: ZetkinOrganizerAction[]): number => {
+    if (!hasActions(v)) {
+      return 2;
```
Why return 2? Sorting comparator functions should return a negative number, zero, or a positive number to indicate the relationship between the compared values (`v1` and `v2` in this case).

The size of the returned value does not matter (except zero), only whether it's negative or positive.

Returning 1 in some execution branches and 2 in other branches makes me think that it matters somehow, but it shouldn't, so that makes me think that it's a mistake. In any case, it makes me think, which makes reading more difficult. 😊 

Would it make sense to change this?

```suggestion
      return 1;
```

helihard commented:
I can see where I made the code unnecessary complicated and less readable. I have removed the named variable, since in the end I decided on it being unnecessary, it is clear enough without it what the code does. I have also removed my public display of affection for TS syntax.

@richardolsson: You are, to no surprise, right. What I wanted to achieve was to get the priority of each of the three cases (orange, gray and white), where those that do not have an array of organizer actions in the payload get the lowest priority, in this case 2. The helper (renamed for clarity) returns either 2, 1 or 0, while the sorter should return a positive, negative or 0 by comparing the priorities.

richardolsson commented:
> @richardolsson: You are, to no surprise, right. What I wanted to achieve was to get the priority of each of the three cases (orange, gray and white), where those that do not have an array of organizer actions in the payload (whites) get the lowest priority, in this case 2. The helper (renamed for clarity) returns either 2, 1 or 0, while the sorter should return a positive, negative or 0 by comparing the priorities.

I actually think I was wrong, and you were right. I didn't realize when I read the code that this was an inner helper function that had a reason for returning 0, 1 or 2. I think your reasoning is good! 💯 

That said, I haven't had time to look at the recent changes. I think @ziggabyte will finish the review!

ziggabyte reviewed (approved):
We are ready for takeoff!!! 🚀 ⭐
